gordo_components/machine/metadata/metadata.py

```diff
@@ -0,0 +1,42 @@
+from typing import Dict, Any, Optional
+from dataclasses import dataclass, field
+from dataclasses_json import dataclass_json
+from gordo_components import __version__
+
+
+__all__ = ["Metadata", "BuildMetadata", "ModelBuildMetadata", "CrossValidationMetaData"]
+
+
+@dataclass_json
+@dataclass
+class CrossValidationMetaData:
+    scores: Dict[str, Any] = field(default_factory=dict)
+    cv_duration_sec: Optional[float] = field(default=None)
+    splits: Dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass_json
+@dataclass
+class ModelBuildMetadata:
```

Comment:
It seems like once when we create these they could be immutable after that, in which case you could use the [`frozen`](https://docs.python.org/3/library/dataclasses.html#module-level-decorators-classes-and-functions) arg to the decorator.

Reply:
As we talked about, lower branches of the metadata such as `CrossValMetadata` and similar could maybe, in some way, benefit from `frozen` but it wasn't at the top level nodes because we naturally (as of now) inject updates to the metadata as it flows through the process.

Reply:
Though I'm not strongly opinionated about it either, it make sense to me. I would think that we probably don't want to alter the metadata for the cross-validation after it's occurred (for example), so if there is something that needs to be mutable down the line it would be good to have it made so explicitly. We can always revisit this later.

Reply:
Yep, like we talked about, seems dataclass inherit their owner's frozen setting. Maybe we could revisit this later if we really want to make the whole thing frozen.